ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/export/elements/GeneratorUcteExport.java

```diff
@@ -0,0 +1,125 @@
+package com.powsybl.ucte.converter.export.elements;
+
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.ReactiveLimits;
+import com.powsybl.ucte.network.UcteNode;
+import com.powsybl.ucte.network.UcteNodeTypeCode;
+import com.powsybl.ucte.network.UctePowerPlantType;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static com.powsybl.ucte.converter.util.UcteConverterConstants.DEFAULT_POWER_LIMIT;
+import static com.powsybl.ucte.converter.util.UcteConverterConstants.POWER_PLANT_TYPE_PROPERTY_KEY;
+
+/**
+ * @author Coline Piloquet {@literal <coline.piloquet at rte-france.com>}
+ */
+public final class GeneratorUcteExport {
+
+    private GeneratorUcteExport() {
+    }
+
+    /**
+     * Initialize the power generation fields from the generators connected to the specified bus.
+     *
+     * @param ucteNode The UCTE node to fill
+     * @param bus The bus the generators are connected to
+     */
+    public static void convertGenerators(UcteNode ucteNode, Bus bus) {
+        double activePowerGeneration = 0;
+        double reactivePowerGeneration = 0;
+        List<Double> voltageReferences = new ArrayList<>();
+        List<Double> minPs = new ArrayList<>();
+        List<Double> maxPs = new ArrayList<>();
+        List<Double> minQs = new ArrayList<>();
+        List<Double> maxQs = new ArrayList<>();
+
+        UcteNodeTypeCode nodeType = UcteNodeTypeCode.PQ;
+        List<UctePowerPlantType> powerPlantTypes = new ArrayList<>();
+
+        for (Generator generator : bus.getGenerators()) {
+            if (!Double.isNaN(generator.getTargetP())) {
+                activePowerGeneration += generator.getTargetP();
+            }
+            if (!Double.isNaN(generator.getTargetQ())) {
+                reactivePowerGeneration += generator.getTargetQ();
+            }
+            if (generator.isVoltageRegulatorOn()) {
+                // If one of the generators regulates voltage, then the node is a PU node.
+                nodeType = UcteNodeTypeCode.PU;
+                if (!Double.isNaN(generator.getTargetV())) {
+                    voltageReferences.add(getTargetV(generator));
```

Comment:
There was no check before that the generator is regulating. But I don't think it makes sense to fill the value if there is no regulation.